File: leetcode/1346-check-if-n-and-its-double-exist/solution.cpp
/**
 * Author: 1chooo<hugo970217@gmail.com>
 * Problem: https://leetcode.com/problems/check-if-n-and-its-double-exist/
 * Runtime: ms (%)
 */

const static auto _ = []() {
    cin.tie(nullptr)->sync_with_stdio(false);
    return nullptr;
}();

class Solution {
  public:
    bool checkIfExist(vector<int> &arr) {
        unordered_set<int> seen;
        for (int num : arr) {
            if (seen.count(num * 2) ||
                (num % 2 == 0 && seen.count(num / 2))) {
                return true;
            }
            seen.insert(num);
        }
        return false;
    }
};

File: leetcode/1346-check-if-n-and-its-double-exist/solution.py
"""
* Author: 1chooo<hugo970217@gmail.com>
* Problem: https://leetcode.com/problems/check-if-n-and-its-double-exist/
* Runtime: 0ms (100.00%)
"""

class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        seen: set = set()

        for num in arr:
            if (2 * num in seen or 
                (num % 2 == 0 and num // 2 in seen)):
                return True
            seen.add(num)

        return False
